### ISSUE TITLE
Error: Variable declaration in a .pkrvar file (Packer v1.11.2)

### ISSUE THREAD

**Issue by aenagy** [brandonleegit/PackerBuilds#6]
While trying to follow the procedure in the YouTube video (https://www.youtube.com/watch?v=UYQX7fW31jU) I modified the files for Windows Server 2019:

```
C:\Users\gitlabUsername\git\fqdnOfMyGitlabServer\gitlabUsername\WindowsServer2019_packer\HCL_build> packer build -var-file .\variables.pkr.hcl .\WindowsServer2019.json.pkr.hcl
Error: Variable declaration in a .pkrvar file

  on .\variables.pkr.hcl line 1, in variable "cpu_num":
   1: variable "cpu_num" {

A .pkrvar file is used to assign values to variables that have already been
declared in .pkr files, not to declare new variables. To declare variable
"cpu_num", place this block in one of your .pkr files, such as variables.pkr.hcl

To set a value for this variable in .\variables.pkr.hcl, use the definition
syntax instead:
    cpu_num = <value>

Error: Variable declaration in a .pkrvar file

  on .\variables.pkr.hcl line 6, in variable "disk_size":
   6: variable "disk_size" {

A .pkrvar file is used to assign values to variables that have already been
declared in .pkr files, not to declare new variables. To declare variable
"disk_size", place this block in one of your .pkr files, such as
variables.pkr.hcl

To set a value for this variable in .\variables.pkr.hcl, use the definition
syntax instead:
    disk_size = <value>
<snip>
```

I haven't included all of the error output but the pattern is the same as shown here. The contents of `variables.pkr.hcl` have not been changed since cloning from this repo.

What is causing this failure?
What is the fix?

**Comment by aenagy** [comment]
Because I am new to packer some things are not obvious.

First, I pulled out the `vsphere_user`, `vsphere_password` and `winadmin_password` from the file `windowsserver2019.auto.pkrvars.hcl` and put it into a new file `secrets.hcl` in the same folder. I did this so that in my `.gitignore` file I could add `*secrets*.*` and eliminate the possibility of accidently pushing credentials to my Git server.

Second, while the video (https://www.youtube.com/watch?v=UYQX7fW31jU) shows the use of the parameter `-var-file` it seems that this is only necessary for the newly created `secrets.hcl` file.

Third, while researching #2 I discovered (
![image](https://github.com/user-attachments/assets/291ed15e-283b-4f33-b3c8-97dbba5c880e)
) the correct syntax to use was to simply specify the location of the `hcl` files.

Working solution:

`packer build -var-file .\secrets.hcl .`

Note the trailing dot indicating that packer should use the `hcl` files in the current folder.
